src/transformers/modeling_utils.py

```diff
@@ -2299,6 +2323,10 @@ def from_pretrained(cls, pretrained_model_name_or_path: Optional[Union[str, os.P
         with ContextManagers(init_contexts):
             model = cls(config, *model_args, **model_kwargs)
 
+        keep_in_fp32_modules = model._keep_in_fp32_modules
+        if keep_in_fp32_modules is not None:
+            low_cpu_mem_usage = True
```

Comment:
Shouldn't this use `use_keep_in_fp32_modules` here? Also should go before so the test at line 2307 can be simplified.

Reply:
I simplified the tests in 966cc06 but I think that we still need `keep_in_fp32_modules = model._keep_in_fp32_modules` as it is used later on in line 2342

Reply:
I think you are right: 0f75387

Reply:
Actually it seems that it's more tricky than that, putting it on top will result in some failing tests should be fixed in 243e6b5